src/code_8097504.c

```diff
@@ -4,29 +4,25 @@
 #include "dungeon.h"
 #include "event_flag.h"
 #include "code_80A26CC.h"
+#include "code_80972F4.h"
 
 ALIGNED(4) const char gMeetNinetalesText[] =  "Meet Ninetales.";
 ALIGNED(4) const char gAvoidCaptureText[] = "Avoid capture.";
 ALIGNED(4) const char gFinalScenarioText[] = _("Defeat the final Pokémon.");
 
+// Needed to match
+static inline s16 Self(s16 a)
```

Comment:
This smells similar to the `SpeciesId()` inline but instead it's for maze ids

Reply:
Yeah, it makes me wonder if `SpeciesId` should actually exist and there's some unholy `InlineCastToS16` used instead

Reply:
Nah I think it is definitely a SpeciesId but there also happen to be 2 other times they do that kind of conversion (like maze/friendarea and maybe item/move or something). We'll keep finding them

Reply:
True, but there are issues all around the codebase with s16 arguments which don't match. So unless they had conversion used for everything, it's possible they performed some weird inline cast for all s16. 
Though some of them are unknown arguments, so maybe they're indeed just species/moves/items/etc.